docs/src/main/asciidoc/kafka-schema-registry-avro.adoc

```diff
@@ -0,0 +1,617 @@
+  http://localhost:8080/movies
+
+curl --header "Content-Type: application/json" \
+  --request POST \
+  --data '{"title":"The Godfather","year":1972}' \
+  http://localhost:8080/movies
+
+curl --header "Content-Type: application/json" \
+  --request POST \
+  --data '{"title":"The Dark Knight","year":2008}' \
+  http://localhost:8080/movies
+
+curl --header "Content-Type: application/json" \
+  --request POST \
+  --data '{"title":"12 Angry Men","year":1957}' \
+  http://localhost:8080/movies
+----
+
+Observe what is printed in the second terminal. You should see something along the lines of:
+[source]
+----
+data:'The Shawshank Redemption' from 1994
+
+data:'The Godfather' from 1972
+
+data:'The Dark Knight' from 2008
+
+data:'12 Angry Men' from 1957
+----
+
+== Building a native executable
+Building a native executable
+You can build a native executable with the usual command ./mvnw package -Dnative.
+
+Running it is as simple as executing `./target/kafka-avro-schema-quickstart-1.0.0-SNAPSHOT-runner`.
+
+== Testing the application
+
+=== Infrastructure for tests
+We will now use Testcontainers to set up Kafka and Apicurio Schema Registry for tests.
+
+First, let's add test dependencies on Awaitility and Strimzi to `pom.xml`, Testcontainers will be pulled
+in transitively by `strimzi-test-container`:
+[source,xml]
+----
+<dependencies>
+    ...
+    <dependency>
+        <groupId>io.strimzi</groupId>
+        <artifactId>strimzi-test-container</artifactId>
+        <version>0.22.1</version>
+        <scope>test</scope>
+        <exclusions>
+            <exclusion>
+                <groupId>org.apache.logging.log4j</groupId>
+                <artifactId>log4j-core</artifactId>
+            </exclusion>
+        </exclusions>
+    </dependency>
+    <dependency>
+        <groupId>org.awaitility</groupId>
+        <artifactId>awaitility</artifactId>
+        <scope>test</scope>
+    </dependency>
+</dependencies>
+----
+
+Now, let's define a link:getting-started-testing.adoc#quarkus-test-resource[QuarkusTestResourceLifecycleManager] that will create the appropriate
+containers:
+[source,java]
+----
+package org.acme.kafka;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.testcontainers.containers.GenericContainer;
+
+import io.quarkus.test.common.QuarkusTestResourceLifecycleManager;
+import io.strimzi.StrimziKafkaContainer;
+
+public class KafkaAndSchemaRegistryTestResource implements QuarkusTestResourceLifecycleManager {
+
+    private final StrimziKafkaContainer kafka = new StrimziKafkaContainer();
+
+    private GenericContainer<?> registry;
+
+    @Override
+    public Map<String, String> start() {
+        kafka.start();
+        registry = new GenericContainer<>("apicurio/apicurio-registry-mem:1.2.2.Final")
+                .withExposedPorts(8080)
+                .withEnv("QUARKUS_PROFILE", "prod")
+                .withEnv("KAFKA_BOOTSTRAP_SERVERS", kafka.getBootstrapServers())
```

Comment:
the env vars `KAFKA_BOOTSTRAP_SERVERS` , `APPLICATION_ID` and `APPLICATION_SERVER` are not required, the image `apicurio/apicurio-registry-mem` stores the schemas in-memory and do not require access to kafka

Reply:
Good point. I'm updating the guide to use Apicurio Registry 2.0.0, so I'll remove this as part of the work.